Tests/SonarRunner.Shim.Tests/SonarRunnerWrapperTests.cs

```diff
@@ -171,20 +167,85 @@ public void SonarRunner_SensitiveArgsPassedOnCommandLine()
             Assert.IsTrue(userPwdIndex < standardArgsIndex && userPwdIndex < propertiesFileIndex, "User arguments should appear first");
         }
 
+        [TestMethod]
+        public void WrapperError_Success_NoStdErr()
+        {
+            TestWrapperErrorHandling(exitCode: SuccessExitCode, addMessageToStdErr: false, expectedOutcome: true);
+        }
+
+        [TestMethod]
+        [WorkItem(202)] //SONARMSBRU-202
+        public void WrapperError_Success_StdErr()
+        {
+            TestWrapperErrorHandling(exitCode: SuccessExitCode, addMessageToStdErr: true, expectedOutcome: true);
+        }
+
+        [TestMethod]
+        public void WrapperError_None_NoStdErr()
+        {
+            TestWrapperErrorHandling(exitCode: null, addMessageToStdErr: false, expectedOutcome: true);
+        }
+
+        [TestMethod]
+        public void WrapperError_None_StdErr()
+        {
+            TestWrapperErrorHandling(exitCode: null, addMessageToStdErr: true, expectedOutcome: true);
+        }
+
+        [TestMethod]
+        public void WrapperError_Fail_NoStdErr()
+        {
+            TestWrapperErrorHandling(exitCode: FailureExitCode, addMessageToStdErr: false, expectedOutcome: false);
+        }
+
+        [TestMethod]
+        public void WrapperError_Fail_StdErr()
+        {
+            TestWrapperErrorHandling(exitCode: FailureExitCode, addMessageToStdErr: true, expectedOutcome: false);
+        }
+
+        private void TestWrapperErrorHandling(int? exitCode, bool addMessageToStdErr, bool expectedOutcome)
+        {
+            // Arrange
+            TestLogger logger = new TestLogger();
+            string exePath = CreateDummarySonarRunnerBatchFile(addMessageToStdErr, exitCode);
+            string propertiesFilePath = CreateDummySonarRunnerPropertiesFile();
+            AnalysisConfig config = new AnalysisConfig() { SonarRunnerWorkingDirectory = this.TestContext.DeploymentDirectory };
+
+            // Act
+            bool success = SonarRunnerWrapper.ExecuteJavaRunner(config, Enumerable.Empty<string>(), logger, exePath, propertiesFilePath);
+
+            // Assert
+            VerifyProcessRunOutcome(logger, this.TestContext.DeploymentDirectory, success, expectedOutcome);
+        }
+
         #endregion Tests
 
         #region Private methods
 
-        private string CreateDummarySonarRunnerBatchFile()
+        private string CreateDummarySonarRunnerBatchFile(bool addMessageToStdErr = false, int? exitCode = null)
```

Comment:
Not much point in allowing nulls for the exit code since the batch file will return 0 in this case.